scout/server/blueprints/variants/static/form_scripts.js

```diff
@@ -1,69 +1,107 @@
-  var matchedChrName = chromPos.match(chromPosPattern)
-  if(chrom==="" && chromPos===""){
-    startElem.value = "";
-    endElem.value = "";
-    return //only reset cytoband select element
-  // Set the selected chromosome name
-  } else if (chrom !== "" && matchedChrName !== null){
-    chromosome = matchedChrName[1]
-  } else if (chrom === "" && matchedChrName !== null){
-    chromosome = matchedChrName[1]
-  } else if (chrom == "" && matchedChrName == null) {
-    return
-  } else {
-    chromosome = chrom;
-  }
+	var chrom_select = document.forms["filters_form"].elements["chrom"];
+  console.log("chrom select has type " + typeof(chrom_select))
 
-  var chrom_cytobands = cytobands[chromosome]["cytobands"]; // chromosome-specific cytobands
+	var chrom = [];
+  var options = chrom_select && chrom_select.options;
+  var opt;
 
-  for (elem of [cytoStart, cytoEnd]) {
-    if (elem.options.length > 0){
-      elem.options.length = 0; //remove previous select options
+  for (var i=0, iLen=options.length; i<iLen; i++) {
+    opt = options[i];
+
+    if (opt.selected) {
+    	if (opt.value !== "") {
+				chrom.push(opt.value);
+			}
     }
-    var emptyStart = document.createElement("option");
-    emptyStart.textContent = "";
-    emptyStart.value = "";
-    elem.appendChild(emptyStart); //Add an empty (blank) option to the select
   }
 
-  for(var i = 0; i < chrom_cytobands.length; i++) {
-    var opt = chrom_cytobands[i]
+	var chromPos = "";
+	if (typeof document.forms["filters_form"].elements["chrom_pos"] !== "undefined") {
+		chromPos = document.forms["filters_form"].elements["chrom_pos"].value;
+	}
 
-    // populate the cytoband start select
-    var interval = ["(start:",opt["start"], ")"].join("");
-    var optionText = [ chrom, opt["band"], interval ].join(" ");
+	var chromosome = "";
+	console.log("Populate cytobands")
+	var matchedChrName = chromPos.match(chromPosPattern)
+	if (chrom === [] && chromPos === "") {
+		startElem.value = "";
+		endElem.value = "";
+		return //only reset cytoband select element
+		// Set the selected chromosome name
+	} else if (chrom !== [] && matchedChrName !== null) {
+		chromosome = matchedChrName[1]
+	} else if (chrom === [] && matchedChrName !== null) {
+		chromosome = matchedChrName[1]
+	} else if (chrom == [] && matchedChrName == null) {
+		console.log("chrom is empty still")
+		return
+	} else if (typeof chrom === 'string' || chrom instanceof String) {
+		chromosome = chrom;
+	} else {
+		console.log("chrom is array: add multiple chr cytobands")
+		first = true
+		for (var i = 0; i < chrom.length; i++) {
+			if (chrom[i] === undefined) {
+				console.log("chrom " + i + " is undefined!")
+			}
+			console.log("chrom is array: add multiple chr cytobands. first " + i + ":" + chrom[i] + "("+ typeof (chrom[i])+")")
+			if (i !== 0) {
+				first = false
+			}
+			populateCytobandsSingleChr(cytobands, chrom[i], first)
+		}
+		return
+	}
+	populateCytobandsSingleChr(cytobands, chromosome, false)
```

Comment:
Here you changed the populateCytoband by adding possible cytoband options for every chromosome that is preselected in the chromosome select. 

For instance if chromosome 1 and 2 are selected, then all cytobands for chr 1 and 2 are selectable. I understand how you reasoned but perhaps here codefactor has a point: aren't we complicating things too much?

Why not simply inactivating the cytoband select instead? Because your solution is not making it possible to select more than one cytoband at the time anyway, so even if you have 2 chromosome available, your cytoband selection is de facto restricting the search to only one chromosome..